metadata/metadata_test.go

```diff
@@ -205,7 +205,8 @@ func (s) TestFromIncomingContext(t *testing.T) {
 	)
 	// Verify that we lowercase if callers directly modify md
 	md["X-INCORRECT-UPPERCASE"] = []string{"foo"}
-	ctx := NewIncomingContext(context.Background(), md)
+	ctx, cancel := context.WithTimeout(NewIncomingContext(context.Background(), md), defaultTestTimeout)
```

Comment:
Same comment here. I think assigning the timeout higher up in the test would make it more obvious what we are testing.

For those tests that don't do anything with the timeouts, you could even go with a `var` shared for all tests.

Reply:
Fixed. I moved the context with timeout creation to var as you suggested. 
I was not sure if using defer cancel() in each test even tough cancel is only created once is a problem. 
Didn't find any content against it so went with it. Let me know if there is a better way to do it.